src/main/java/com/javatemplate/domain/auth/JwtUserDetailsService.java

```diff
@@ -0,0 +1,44 @@
+package com.javatemplate.domain.auth;
+
+import com.javatemplate.domain.role.Role;
+import com.javatemplate.persistent.role.RoleStore;
+import com.javatemplate.persistent.user.UserEntity;
+import com.javatemplate.persistent.user.UserStore;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+import static com.javatemplate.persistent.user.UserEntityMapper.toUserEntity;
+import static java.lang.String.format;
+
+@Service
+@RequiredArgsConstructor
+public class JwtUserDetailsService implements UserDetailsService {
+
+    private final UserStore userStore;
+
+    private final RoleStore roleStore;
+
+    @Override
+    public UserDetails loadUserByUsername(final String username) throws UsernameNotFoundException {
+        return userStore.findByUsername(username)
+                .map(user -> buildUser(toUserEntity(user)))
+                .orElseThrow(() -> new UsernameNotFoundException(format("User not found with username: %s", username)));
```

Comment:
```suggestion
                .orElseThrow(() -> new UsernameNotFoundException(username));
```

Đưa cái message sang bên class Exception

Reply:
Vậy là class này em tạo riêng hay là chỉ dùng của spring security thôi anh

![image](https://user-images.githubusercontent.com/87653000/227097808-2547c6ea-cdaa-4618-aaec-e47345489db1.png)